### ISSUE TITLE
[Feature Request] Add "reduction" parameters when more than 1 critic algorithm as "min" operator is not always the best choice

### ISSUE THREAD

**Issue by jamartinh** [DLR-RM/stable-baselines3#1477]
### 🚀 Feature

Add "reduction" parameters when more than 1 critic algorithm as "min" operator is not always the best choice

### Motivation

min is not always the *best" operator when an algo, e.g. SAC,  has two or more critics.

For some problems "mean" or even "max" is better.

I guess the authors use "min" in the paper assuming 0 to positive rewards, and this is not a standard constraint of defining rerwad functions.

### Pitch

Just add a parameter to specify "critics_reduction_method"  to being able to specifiy it.

### Additional context

1. Monkey patching the current implementation
2. Subclass and modify just one method to have a new algorithm just for this issue.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo

**Comment by araffin** [comment]
Hello,

> min is not always the *best" operator when an algo, e.g. SAC, has two or more critics.


Do you have quantitative experiments to support this claim?

I've tested in the recent past mean instead of min (when playing with DroQ) but couldn't notice any significant difference (one parameter that i would like to add though is a different learning rate for the qf).

The original reason for min comes from the TD3 paper, and it was to reduce overestimation.

> Subclass and modify just one method to have a new algorithm just for this issue.

For now, that would be the way to go, or have a fork of SB3 for experimentations.

**Comment by jamartinh** [comment]
Do you have quantitative experiments to support this claim?
Yes.

I have tested the same issue in two other libs as well, d3rlpy and tianshou
with same consistent result.

I have a gym env where reward goes from -Inf to 0 with Just little special
states where reward is +2 but there are rare.

In this setting, I get consistently a convergece rate with MIN up to some
point where MIN starts to diverge.

If instead of MIN I use MEAN the convergence is:
Faster.
Significant less td error.
Stable.
Actor loss much better.

Indeed I can start experiments with MIN, save the state, and resume with
MEAN instead and after Just some few iterations the td error lowers a lot,
the actor loss improves respectively and the reward improves a lot.

Just have to check using MAX in this setting to verífy if in this setting
is MAX an even better choice.

Best,
Jose

**Comment by araffin** [comment]
> Sorry, what more information is needed?

The hyperparameters used and your system/lib information (os, gym version, mujoco version, sb3 version, ...)

**Comment by araffin** [comment]
> Do you have quantitative experiments to support this claim?

sorry, I meant no custom environment 

> I have a gym env where reward goes from -Inf to 0 with Just little special
states where reward is +2 but there are rare.

> In this setting, I get consistently a convergece rate with MIN up to some
point where MIN starts to diverge.


Looks like you might have an underestimation issue. Have you tried setting `n_critics=1` (so not using any aggregation at all) and have you tried using TQC (from sb3 contrib, you can regulate over/underestimation by truncating a different number of quantiles).

By diverge, do you mean some values like critic loss start diverging?
Might also be related to https://github.com/DLR-RM/rl-baselines3-zoo/issues/322#issuecomment-1345490671 (so trying AdamW with default regularization value might help)
Do you have online runs to share? (best would be with W&B or similar tool)
